src/nifreeze/viz/motion_viz.py

```diff
@@ -0,0 +1,267 @@
+    else:
+        return np.rot90(img_data[slice_idx, :, :])
+
+
+def plot_framewise_displacement(
+    fd1: np.ndarray,
+    fd2: np.ndarray,
+    label1: str,
+    label2: str,
+    ax: Union[Axes, None] = None,
+) -> Axes:
+    """Plot frame-wise displacement data.
+
+    Plots the frame-wise displacement data corresponding to two different
+    realizations.
+
+    Parameters
+    ----------
+    fd1 : :obj:`~numpy.ndarray`
+        Frame-wise displacement values corresponding to first realization.
+    fd2 : :obj:`~numpy.ndarray`
+        Frame-wise displacement values corresponding to second realization.
+    label1 : :obj:`str`
+        Label for the first realization legend.
+    label2 : :obj:`str`
+        Label for the second realization legend.
+    ax : :obj:`Axes`, optional
+        Figure axes.
+
+    Returns
+    -------
+    ax : :obj:`Axes`
+        Figure plot axis.
+    """
+
+    fd1_dims = fd1.ndim
+    fd2_dims = fd2.ndim
+
+    n_frames_fd1 = fd1.shape
+    n_frames_fd2 = fd2.shape
+
+    req_dims = 1
+    if fd1_dims != req_dims or fd2_dims != req_dims:
+        raise ValueError(f"fd1 and fd2 must be 1D: {n_frames_fd1}; {n_frames_fd2}")
+
+    if n_frames_fd1 != n_frames_fd2:
+        raise ValueError(f"fd1 and fd2 must have same shape: {n_frames_fd1}; {n_frames_fd2}")
+
+    if ax is None:
+        fig, ax = plt.subplots(1, 1, figsize=(10, 6), constrained_layout=True)
+
+    # Plot the framewise displacement
+    n_frames = np.arange(len(fd1))
+    ax.plot(n_frames, fd1, label=label1, color="blue")
+    ax.plot(n_frames, fd2, label=label2, color="orange")
+    ax.set_ylabel("FD (mm)")
+    ax.legend(loc="upper right")
+    ax.set_xticks([])  # Hide x-ticks to keep x-axis clean
+
+    return ax
+
+
+def plot_volumewise_motion(
+    frames: np.ndarray,
+    motion_params: np.ndarray,
+    ax: np.ndarray | None = None,
+) -> np.ndarray:
+    """Plot mean volume-wise motion parameters.
+
+    Plots the mean translation and rotation parameters along the ``x``, `y``,
+    and ``z`` axes.
+
+    Parameters
+    ----------
+    frames : :obj:`~numpy.ndarray`
+        Frame indices.
+    motion_params : :obj:`~numpy.ndarray`
+        Motion parameters.Motion parameters: translation and rotation. Each row
+        represents one frame, and columns represent each coordinate axis ``x``,
+        `y``, and ``z``. Translation parameters are followed by rotation
+        parameters column-wise.
+    ax : :obj:`~numpy.ndarray`, optional
+        Figure axes.
+
+    Returns
+    -------
+    ax : :obj:`~numpy.ndarray`
+        Figure plot axes array.
+    """
+
+    if ax is None:
+        fig, ax = plt.subplots(2, 1, figsize=(10, 6), sharex=True, constrained_layout=True)
+
+    # Plot translations
+    ax[0].plot(frames, motion_params[:, 0], label="x")
+    ax[0].plot(frames, motion_params[:, 1], label="y")
+    ax[0].plot(frames, motion_params[:, 2], label="z")
+    ax[0].set_ylabel("Translation (mm)")
```

Comment:
Here Bastiani et al. 2019 use std values; maybe labels could be parameters, or we should compute std values inside the function?